File: day-2/solution_part1.lua
local BUFSIZE = 2^13

local file = io.open("input.txt", "r")
io.input(file)
local file_content = io.read(BUFSIZE, "*line")
io.close(file)

local paper_tobuy = 0

for i in string.gmatch(file_content, "%S+") do
  local values = {}
  local cnt = 0

  for m in string.gmatch(i, "%x+") do
    values[cnt] = m
    cnt = cnt + 1
  end

  local x = values[0] * values[1]
  local y = values[1] * values[2]
  local z = values[0] * values[2]

  local smallest_side = x
  if y < smallest_side then smallest_side = y end
  if z < smallest_side then smallest_side = z end
  paper_tobuy = paper_tobuy + (2 * x) + (2 * y) + (2 * z) + smallest_side
end
print("Paper to buy: " .. paper_tobuy .. " feet.")

File: day-2/solution_part1.py
# length l
# width w
# height h
# formula -> 2*l*w + 2*w*h + 2*h*l + the area of the smallest side

f = open("input.txt", "r").readlines()

total_area = 0
for line in f:
    dimensions = line.split('x')
    dimensions = [float(dimensions[0]), float(dimensions[1]), float(dimensions[2])]

    l, w, h = dimensions[0], dimensions[1], dimensions[2]

    side1 = l * w
    side2 = w * h
    side3 = h * l

    smallest_side = float(min([side1, side2, side3]))

    surface_area = (2 * side1) + (2 * side2) + (2 * side3)

    total_area += surface_area + smallest_side

print('The elves should order ' + str(total_area) + ' square feet of wrapping paper.')
